### PR TITLE
TYP: changed variable new_pd_index to final_pd_index 

### PR DESCRIPTION
**Pull request by prithvijitguha**
xref #37715

Changed variable new_pd_index to final_pd_index in line 2099

Handled mypy type ignore[assignment] error in pandas/io/pytables.py in line 2099

### CODE REVIEW

File: pandas/io/pytables.py [hunk 1]
```diff
@@ -2096,8 +2096,8 @@ def convert(self, values: np.ndarray, nan_rep, encoding: str, errors: str):
 
         # error: Incompatible types in assignment (expression has type
         # "Union[ndarray, DatetimeIndex]", variable has type "Index")
```
Comment (MarcoGorelli) [inline pandas/io/pytables.py:2098]: Thakns @prithvijitguha - this comment can be removed as well, right?

Comment (prithvijitguha) [reply pandas/io/pytables.py:2098]: Yes. its from the previous "#type ignore". Do I need to fix that and commit it? Thanks for reviewing

Comment (prithvijitguha) [reply pandas/io/pytables.py:2098]: @MarcoGorelli Thank you, just removed the comment and removed any remaining trailing whitespace. Should pass now!

**Comment by prithvijitguha** [comment]
Hi @MarcoGorelli Thanks for the review. Its failing one test which is similar to the test failed in another similar pull request #44231
As per this comment, https://github.com/pandas-dev/pandas/pull/44373#issuecomment-964617916 its related to pyarrow. 
What do you suggest? 🤔

**Comment by MarcoGorelli** [comment]
It's unrelated, don't worry about it

This looks good to me, I'm just not sure about the `final_pd_index` variable name, so leaving this open in case others have comments

**Comment by jreback** [comment]
thanks @prithvijitguha

**Pull request merged**